src/wallet/tx_builder.rs

```diff
@@ -560,49 +560,79 @@ impl<'a, B, D: BatchDatabase, Cs: CoinSelectionAlgorithm<D>> TxBuilder<'a, B, D,
-        self.params.recipients.clear();
+    ///
+    /// # Example
+    ///
+    /// `drain_to` is very useful for draining all the coins in a wallet with [`drain_wallet`] to a
+    /// single address.
+    ///
+    /// ```
+    /// # use std::str::FromStr;
+    /// # use bitcoin::*;
+    /// # use bdk::*;
+    /// # use bdk::wallet::tx_builder::CreateTx;
+    /// # let to_address = Address::from_str("2N4eQYCbKUHCCTUjBJeHcJp9ok6J2GZsTDt").unwrap();
+    /// # let wallet = doctest_wallet!();
+    /// let mut tx_builder = wallet.build_tx();
+    ///
+    /// tx_builder
+    ///     // spend all outputs in this wallet
+    ///     .drain_wallet()
+    ///     // send the excess (which is all the coins minus the fee) to this address
+    ///     .drain_to(to_address.script_pubkey())
+    ///     .fee_rate(FeeRate::from_sat_per_vb(5.0))
+    ///     .enable_rbf();
+    /// let (psbt, tx_details) = tx_builder.finish()?;
+    /// # Ok::<(), bdk::Error>(())
+    /// ```
+    ///
+    /// [`maintain_drain_to`]: Self::maintain_drain_to
+    /// [`add_recipient`]: Self::add_recipient
+    /// [`drain_wallet`]: Self::drain_wallet
+    pub fn drain_to(&mut self, script_pubkey: Script) -> &mut Self {
+        self.params.drain_to = Some(script_pubkey);
 
         self
     }
 }
 
 // methods supported only by bump_fee
 impl<'a, B, D: BatchDatabase> TxBuilder<'a, B, D, DefaultCoinSelectionAlgorithm, BumpFee> {
-    /// Bump the fees of a transaction made with [`set_single_recipient`](Self::set_single_recipient)
-    ///
-    /// Unless extra inputs are specified with [`add_utxo`], this flag will make
-    /// `bump_fee` reduce the value of the existing output, or fail if it would be consumed
-    /// entirely given the higher new fee rate.
+    /// Bump the fee of a transaction made with [`drain_to`]
     ///
-    /// If extra inputs are added and they are not entirely consumed in fees, a change output will not
-    /// be added; the existing output will simply grow in value.
+    /// Unless extra inputs are specified with [`add_utxo`] or [`drain_wallet`], this function will
+    /// reduce the value of the `drain_to` output (removing it completely if necessary).
     ///
-    /// Fails if the transaction has more than one outputs.
+    /// Fails if the `drain_to` script can't be found amoung the `script_pubkey`s of the transaction
+    /// we are bumping.
     ///
+    /// [`drain_to`]: Self::drain_to
     /// [`add_utxo`]: Self::add_utxo
-    pub fn maintain_single_recipient(&mut self) -> Result<&mut Self, Error> {
-        let mut recipients = self.params.recipients.drain(..).collect::<Vec<_>>();
-        if recipients.len() != 1 {
-            return Err(Error::SingleRecipientMultipleOutputs);
+    /// [`drain_wallet`]: Self::drain_wallet
+    pub fn maintain_drain_to(&mut self, drain_to: Script) -> Result<&mut Self, Error> {
+        match self.params.recipients.iter().position(|x| x.0 == drain_to) {
```

Comment:
nit: maybe deconstructing the tuple makes it a bit more readable? like `|(script, _)| script == drain_to`